connector/src/main/scala/com/datastax/spark/connector/cql/Scanner.scala

```diff
@@ -35,21 +39,25 @@ class DefaultScanner (
   }
 
   override def scan[StatementT <: Statement[StatementT]](statement: StatementT): ScanResult = {
-    val rs = session.execute(maybeExecutingAs(statement, readConf.executeAs))
-    val columnMetaData = CassandraRowMetadata.fromResultSet(columnNames, rs, codecRegistry)
-    val prefetchingIterator = new PrefetchingResultSetIterator(rs, readConf.fetchSizeInRows)
-    val rateLimitingIterator = readConf.throughputMiBPS match {
-      case Some(throughput) =>
-        val rateLimiter = new RateLimiter((throughput * 1024 * 1024).toLong, 1024 * 1024)
-        prefetchingIterator.map { row =>
-          rateLimiter.maybeSleep(getRowBinarySize(row))
-          row
-        }
-      case None =>
-        prefetchingIterator
-    }
+    import com.datastax.spark.connector.util.Threads.BlockingIOExecutionContext
 
-    ScanResult(rateLimitingIterator, columnMetaData)
+    val rs = session.executeAsync(maybeExecutingAs(statement, readConf.executeAs))
```

Comment:
do you swallow fetch size on purpose?

Reply:
yes, now the caller is responsible for setting the page size on the statement. In most cases the parameter was already set, I had to add it in one or two places.